File: src/main/scala/im/tox/toktok/app/SizeAnimation.scala
package im.tox.toktok.app

import android.view.View
import android.view.animation.{ Animation, Transformation }

final class SizeAnimation(view: View, newSize: Int, sizeChange: Int) extends Animation {

  private val oldSize = {
    if (sizeChange == 0) {
      view.getWidth
    } else {
      view.getHeight
    }
  }

  protected override def applyTransformation(interpolatedTime: Float, t: Transformation): Unit = {
    val currentSize = oldSize + ((newSize - oldSize) * interpolatedTime).toInt

    sizeChange match {
      case 0 => view.getLayoutParams.width = currentSize
      case 1 => view.getLayoutParams.height = currentSize
    }

    view.requestLayout()
  }

  protected override def willChangeBounds: Boolean = true

}

File: sources/app/src/main/java/im/tox/toktok/app/SizeAnimation.java
package im.tox.toktok.app;

import android.view.View;
import android.view.animation.Animation;
import android.view.animation.Transformation;

import androidx.annotation.NonNull;

public final class SizeAnimation extends Animation {

    @NonNull
    private final View view;
    private final int oldSize;
    private final int newSize;
    private final int sizeChange;

    public SizeAnimation(@NonNull View view, int newSize, int sizeChange) {
        this.view = view;
        this.oldSize = sizeChange == 0 ? view.getWidth() : view.getHeight();
        this.newSize = newSize;
        this.sizeChange = sizeChange;
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {
        final int currentSize = (int) (oldSize + ((newSize - oldSize) * interpolatedTime));

        switch (sizeChange) {
            case 0:
                view.getLayoutParams().width = currentSize;
                break;
            case 1:
                view.getLayoutParams().height = currentSize;
                break;
        }

        view.requestLayout();
    }

    @Override
    public boolean willChangeBounds() {
        return true;
    }

}
